src/main/java/org/jetbrains/research/intellijdeodorant/core/ast/util/StatementExtractor.java

```diff
@@ -1,265 +1,265 @@
-            statementCounter += 1;
-            statementCounter += getTotalNumberOfStatements(switchStatement);
-        } else if (statement instanceof PsiAssertStatement) {
-            statementCounter += 1;
-        } else if (statement instanceof PsiLabeledStatement) {
-            PsiLabeledStatement labeledStatement = (PsiLabeledStatement) statement;
-            statementCounter += 1;
-            statementCounter += getTotalNumberOfStatements(labeledStatement.getStatement());
-        } else if (statement instanceof PsiReturnStatement) {
-            statementCounter += 1;
-        } else if (statement instanceof PsiSynchronizedStatement) {
-            PsiSynchronizedStatement synchronizedStatement = (PsiSynchronizedStatement) statement;
-            statementCounter += 1;
-            statementCounter += getTotalNumberOfStatements(synchronizedStatement);
-        } else if (statement instanceof PsiThrowStatement) {
-            statementCounter += 1;
-        } else if (statement instanceof PsiTryStatement) {
-            PsiTryStatement tryStatement = (PsiTryStatement) statement;
-            statementCounter += 1;
-            statementCounter += getTotalNumberOfStatements(tryStatement);
-            PsiCodeBlock[] catchClauses = tryStatement.getCatchBlocks();
-            for (PsiCodeBlock catchClause : catchClauses) {
-                PsiStatement[] statementArray = catchClause.getStatements();
-                for (PsiStatement statement1 : statementArray) {
-                    statementCounter += getTotalNumberOfStatements(statement1);
-                }
-            }
-            PsiCodeBlock finallyBlock = tryStatement.getFinallyBlock();
-            if (finallyBlock != null) {
-                PsiStatement[] finallyStatements = finallyBlock.getStatements();
-                for (PsiStatement statement1 : finallyStatements)
-                    statementCounter += getTotalNumberOfStatements(statement1);
-            }
-        } else if (statement instanceof PsiDeclarationStatement) {
-            statementCounter += 1;
-        } else if (statement instanceof PsiBreakStatement) {
-            statementCounter += 1;
-        } else if (statement instanceof PsiContinueStatement) {
-            statementCounter += 1;
-        }
-        return statementCounter;
-    }
-}
+package org.jetbrains.research.intellijdeodorant.core.ast.util;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import com.intellij.psi.*;
+
+public class StatementExtractor {
+
+    public StatementExtractor() {
+
+    }
+
+    private StatementInstanceChecker instanceChecker;
+
+    public List<PsiStatement> getConstructorInvocations(PsiStatement statement) {
+        instanceChecker = new InstanceOfConstructorInvocation();
+        return getStatements(statement);
+    }
+
+    public List<PsiStatement> getVariableDeclarationStatements(PsiElement statement) { // TODO: returns all declaration statements
```

Comment:
Is it actual? I have fixed it in master

Reply:
It was an outdated comment, I removed it.